src/Tribe/Assets.php

```diff
@@ -0,0 +1,286 @@
+<?php
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+/**
+ * Class used to register and enqueue assets across our plugins
+ */
+class Tribe__Assets {
+	/**
+	 * Static Singleton Holder
+	 *
+	 * @var self|null
+	 */
+	protected static $instance;
+
+	/**
+	 * Static Singleton Factory Method
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( empty( self::$instance ) ) {
+			self::$instance = new self;
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Register the Methods in the correct places
+	 */
+	private function __construct() {
+		// Hook the actual rendering of notices
+		add_action( 'wp_enqueue_scripts', array( $this, 'hook' ), 1 );
+		add_action( 'admin_enqueue_scripts', array( $this, 'hook' ), 1 );
+		add_action( 'login_enqueue_scripts', array( $this, 'hook' ), 1 );
+	}
+
+	public function hook() {
+		foreach ( $this->assets as $asset ) {
+			if ( 'js' === $asset->type ) {
+				wp_register_script( $asset->slug, $asset->url, $asset->deps, $asset->version, $asset->in_footer );
+			} else {
+				wp_register_style( $asset->slug, $asset->url, $asset->deps, $asset->version, $asset->media );
+			}
+
+			if ( ! is_string( $asset->action ) ) {
+				continue;
+			}
+
+			add_action( $asset->action, array( $this, 'enqueue' ), $asset->priority );
+		}
+	}
+
+	public function enqueue() {
+		foreach ( $this->assets as $asset ) {
+			// Skip if we are not on the correct filter
+			if ( current_filter() !== $asset->action ) {
+				continue;
+			}
+
+			// If here is no action we are just registering this asset
+			if ( ! is_string( $asset->action ) ) {
+				continue;
+			}
+
+			// If we have a set of conditionals we loop on then and get if they are true
+			if ( ! empty( $asset->conditionals ) ) {
```

Comment:
We discussed via Hangouts, but just to be thorough: Let's ditch the conditional stuff in favor of an `apply_filters()` call for a boolean, here.